File: C++/hashmap.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    unordered_map<string, int> mp;

    // Insert key-value pairs
    mp["apple"] = 5;
    mp["banana"] = 2;

    // Access value
    cout << "apple count: " << mp["apple"] << endl;  // Output: 5

    // Check if key exists
    if (mp.find("banana") != mp.end())
        cout << "banana exists\n";

    // Iterate hashmap
    for (auto& pair : mp)
        cout << pair.first << ": " << pair.second << endl;
}

File: Data/Java/hashmap.java
import java.util.*;

public class HashMapExample {

    public static void main(String[] args) {
        // Create a HashMap to store String keys and Integer values
        HashMap<String, Integer> map = new HashMap<>();

        // Put key-value pairs
        map.put("apple", 2);
        map.put("banana", 5);
        map.put("orange", 3);

        System.out.println("HashMap: " + map);  // {banana=5, orange=3, apple=2}

        // Get value using key
        int count = map.get("banana");  // 5
        System.out.println("Bananas: " + count);

        // Check if a key exists
        if (map.containsKey("apple")) {
            System.out.println("We have apples.");  // true
        }
    }
}
